File: src/supabaseClient.js
import { createClient } from '@supabase/supabase-js';

// Retrieve public environment variables
const supabaseUrl = import.meta.env.VITE_SUPABASE_URL;
const supabaseAnonKey = import.meta.env.VITE_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseAnonKey) {
  console.error("Supabase frontend environment variables are missing.");
}

// Initialize the client governed by RLS
export const supabase = createClient(supabaseUrl, supabaseAnonKey);

File: supabase_client.py
import os
from supabase import create_client, Client

# Retrieve environment variables injected by the environment
supabase_url: str = os.environ.get("SUPABASE_URL")
supabase_key: str = os.environ.get("SUPABASE_SERVICE_KEY")

if not supabase_url or not supabase_key:
    raise ValueError("Missing Supabase environment variables in runtime context.")

# Initialize the administrative client
supabase: Client = create_client(supabase_url, supabase_key)
